frontend/src/hooks/useIsAuthorized/useIsAuthorized.ts

```diff
@@ -0,0 +1,13 @@
+import { User } from "@/domain/User";
+
+export const useIsAuthorized = (currentUser: User, authorizedRoles: Array<string>) => {
+    if(currentUser){
+        if(!currentUser.roles){
+            return false;
+        }
+        return currentUser.roles.some((role) => authorizedRoles.includes(role));
+    }
+    else {
+        return false;
+    }
+}
```

Comment:
How about:

```ts
export const isAuthorized = (currentUser: User, authorizedRoles: Array<string>) => {
  if(!currentUser?.roles) return false
  
  return currentUser.roles.some((role) => authorizedRoles.includes(role))
}
```

or
```ts
export const isAuthorized = (role: string, authorizedRoles: Array<string>) => {  
  roles.some((role) => authorizedRoles.includes(role))
}
```

Also, this doesn't need the `use` prefix because technically it isn't a hook, just a regular js function.
Check this on the `deep dive` section: https://react.dev/learn/reusing-logic-with-custom-hooks#hook-names-always-start-with-use

Reply:
I actually meant to delete this hook entirely now that I moved the first check to the middleware. For protecting specific components on a page - like the new 'delete task ' button Leonardo has added - it is a simple check for a scope (`user.authorizedScopes.includes("task:delete-own")`) and I don't think that this needs a separate function now.